Makefile

```diff
@@ -604,6 +612,33 @@ $(SBOM): $(ROOTFS_TAR) | $(INSTALLER)
 	rm -rf $(TMP_ROOTDIR)
 	$(QUIET): $@: Succeeded
 
+$(GOSOURCES):
+	$(QUIET): $@: Begin
+	$(shell GOBIN=$(BUILDTOOLS_BIN)/yash GO111MODULE=on CGO_ENABLED=0 go install $(GOSOURCES_SOURCE)@$(GOSOURCES_VERSION))
+	@echo Done building packages
+	$(QUIET): $@: Succeeded
+
+$(COLLECTED_SOURCES): $(ROOTFS_TAR) $(GOSOURCES)| $(INSTALLER)
+	$(QUIET): $@: Begin
+	$(eval TMP_ROOTDIR := $(shell mktemp -d))
+	# this is a bit of a hack, but we need to extract the rootfs tar to a directory, and it fails if
+	# we try to extract character devices, block devices or pipes, so we just exclude the dir.
+	# when syft supports reading straight from a tar archive with duplicate entries,
+	# this all can go away, and we can read the rootfs.tar
+	# see https://github.com/anchore/syft/issues/1400
+	tar xf $< -C $(TMP_ROOTDIR) --exclude "dev/*"
+	$(eval TMP_COLLECTED_SOURCES := $(shell mktemp -d))
+	bash tools/get-alpine-pkg-source.sh -s $(TMP_COLLECTED_SOURCES)/alpine/ -e $(TMP_ROOTDIR) 
+	bash tools/get-kernel-source.sh -s $(TMP_COLLECTED_SOURCES)/kernel/
+	for i in $$(find . -name 'go.sum'); \
+		do								 		   \
+			go-sources-and-licenses sources -d $$(dirname $$i) --recursive --out $(TMP_COLLECTED_SOURCES)/go; \
```

Comment:
You can just use `$(GOSOURCES)` here, and avoid the path issues:

```
			$(GOSOURCES) sources -d $$(dirname $$i) --recursive --out $(TMP_COLLECTED_SOURCES)/go; \
```

Reply:
fixed